org.mwc.cmap.TimeController/src/org/mwc/cmap/TimeController/recorders/CoordinateRecorder.java

```diff
@@ -144,14 +149,297 @@ protected void showMessageDialog(final String message)
+            }
+          };
+
+      // and start it running
+      reader.start();
+
+      // wait for the results
+      while (reader.isAlive())
+      {
+        try
+        {
+          Thread.sleep(100);
+        }
+        catch (final java.lang.InterruptedException e)
+        {
+          Application.logError2(Application.INFO, "Error while sleeping", e);
+        }
+      }
+
+      /*
+       * final PlainProjection projection = new org.mwc.cmap.gt2plot.proj.GtProjection();
+       */
+      final PlainProjection projection =
+          new MWC.Algorithms.Projections.FlatProjection();
+      projection.setScreenArea(new java.awt.Dimension(1443, 901));
+      projection.setDataArea(new WorldArea(new WorldLocation(22.238965795584505,
+          -21.928244631862952, 0), new WorldLocation(22.238965795584505,
+              -21.43985414608609, 0)));
+
+      final org.mwc.cmap.core.DataTypes.Temporal.TimeControlProperties timePreferences =
+          new org.mwc.cmap.core.DataTypes.Temporal.TimeControlProperties();
+
+      CoordinateRecorder recorder = new CoordinateRecorder(_theLayers,
+          projection, timePreferences);
+
+      return recorder;
+    }
+
+    /**
+     * COLLINGWOOD starts after NELSON
+     */
+    public void testPrimaryStartsFirst()
+    {
+      CoordinateRecorder recorder = getRecorder();
+
+      MWC.GenericData.HiResDate currentTime = new MWC.GenericData.HiResDate(
+          818748540000L);
+
+      final int AMOUNT_OF_STEPS = 3;
+      doIteration(recorder, currentTime, AMOUNT_OF_STEPS);
+
+      final java.util.Map<String, Debrief.ReaderWriter.powerPoint.model.Track> track =
+          recorder._tracks;
+      checkTrackSize(track);
+      assertTrue(track.get(collingwood).getStepsToSkip() == 1);
+      assertTrue(track.get(collingwood).getSegments().size() == 2);
+      assertTrue("0064bd".equals(track.get(collingwood).getColorAsString()));
+      checkCollingwoodSegments(track);
+      assertTrue(track.get(nelson).getSegments().size() == 3);
+      assertTrue("e01c3e".equals(track.get(nelson).getColorAsString()));
+      final Debrief.ReaderWriter.powerPoint.model.TrackPoint firstSegmentNelson =
+          new Debrief.ReaderWriter.powerPoint.model.TrackPoint();
+      firstSegmentNelson.setCourse(2.6267204f);
+      firstSegmentNelson.setLatitude(20.0f);
+      firstSegmentNelson.setLongitude(245.0f);
+      firstSegmentNelson.setSpeed(1.1252066f);
+      firstSegmentNelson.setTime(new Date(818748540000L));
+      firstSegmentNelson.setFormattedTime("120609");
+      assertEquals(firstSegmentNelson, track.get(nelson).getSegments().get(0));
+      final Debrief.ReaderWriter.powerPoint.model.TrackPoint secondSegmentNelson =
+          new Debrief.ReaderWriter.powerPoint.model.TrackPoint();
+      secondSegmentNelson.setCourse(2.6162486f);
+      secondSegmentNelson.setLatitude(14.0f);
+      secondSegmentNelson.setLongitude(250.0f);
+      secondSegmentNelson.setSpeed(1.1252066f);

```

Comment:
@saulhidalgoaular - I'm not a fan of all of the magic numbers in the testing, and am worried that they may become a maintenance burden.  

Mind you, you could argue that if these tests start to fail then it could be an early warning of an issue in some unrelated area.   I don't know which way to swing on that, happy to discuss :-)

Is there a simpler approach?  For this test, where we're validating that we handle tracks starting at different times, I _guess_ that we'll be confident things are working if the tracks have the correct number of children.